### PR TITLE
Provide nicer error message for unsupported LIMIT clause

### PR DESCRIPTION
**Pull request by Holmistr**
Fixes: https://github.com/hazelcast/hazelcast/issues/17294

### CODE REVIEW

File: hazelcast-sql/src/main/java/com/hazelcast/sql/impl/calcite/parse/UnsupportedOperationVisitor.java [hunk 1]
```diff
@@ -165,6 +165,10 @@ private void processSelect(SqlSelect select) {
         if (select.getGroup() != null && select.getGroup().size() > 0) {
             throw unsupported(select.getGroup(), "GROUP BY");
         }
+
+        if (select.getFetch() != null) {
```
Comment (viliam-durina) [inline hazelcast-sql/src/main/java/com/hazelcast/sql/impl/calcite/parse/UnsupportedOperationVisitor.java:169]: We need the same for OFFSET probably

Comment (devozerov) [inline hazelcast-sql/src/main/java/com/hazelcast/sql/impl/calcite/parse/UnsupportedOperationVisitor.java:169]: We need a unit test to cover this case in `ParserOperationsTest`. Something like:
```
@Test
    public void testUnsupportedLimit() {
        checkFailure(
            "SELECT a FROM t LIMIT 10",
            "LIMIT is not supported"
        );
    }
```

Comment (Holmistr) [reply hazelcast-sql/src/main/java/com/hazelcast/sql/impl/calcite/parse/UnsupportedOperationVisitor.java:169]: Fixed.

Comment (Holmistr) [reply hazelcast-sql/src/main/java/com/hazelcast/sql/impl/calcite/parse/UnsupportedOperationVisitor.java:169]: Fixed.

**Pull request merged**